pkg/ens/ens_test.go

```diff
@@ -0,0 +1,86 @@
+package ens_test
+
+import (
+	"encoding/hex"
+	"testing"
+
+	"github.com/paxthemax/resolver/pkg/ens"
+)
+
+const (
+	cloudflareEndpoint = "https://cloudflare-eth.com"
+)
+
+func TestConnect(t *testing.T) {
+	tests := map[string]struct {
+		endpoint   string
+		shouldFail bool
+	}{
+		"fails, empty": {
+			endpoint:   "",
+			shouldFail: true,
+		},
+		"fails, not an eth node": {
+			endpoint:   "https://example.com",
+			shouldFail: true,
+		},
+		"passes": {
+			endpoint: cloudflareEndpoint,
+		},
+	}
+
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			res := ens.NewResolver(tc.endpoint)
+			err := res.Connect()
+			if tc.shouldFail {
+				if err == nil {
+					t.Fatalf("Error: test should fail")
+				}
+				return
+			}
+
+			if !res.IsConnected() {
+				t.Fatalf("Error: isConnected should be true")
+			}
+			wantEndpoint := res.Endpoint()
+			gotEndpoint := tc.endpoint
+			if wantEndpoint != gotEndpoint {
+				t.Fatalf("Error in endpoint: want %v, got %v", wantEndpoint, gotEndpoint)
+			}
+		})
+	}
+
+}
+
+func TestResolve(t *testing.T) {
+	tests := map[string]struct {
+		name       string
+		want       string
+		shouldFail bool
+	}{
+		"passes": {
+			name: "nickjohnson.eth",
```

Comment:
This test case is a bit tricky, as it depends on the external resource that we do not control. As I have no concrete suggestion, except to mock ensclient, this could stay as it is with a comment that it needs to be updated when the domain changes.

Reply:
I think that we should control the domain that is tested. @svetomir can we create one domain only for purpose of testing its record in this case?

Reply:
We can. Do you have any particular domain in mind already?

Reply:
Not really. Any ideas are welcome. Maybe @paxthemax has one in mind?

Reply:
@janos Either we register `swarm-test.eth` as a domain, or `test.swarm.eth`.

Reply:
@paxthemax @janos I registered test.swarm.eth

Reply:
Thanks, test.swarm.eth is cool.

Reply:
@svetomir you beat me to it :D

Reply:
@paxthemax any progress on the testing changes or should we have followup PR for that?